server/src/test/java/org/elasticsearch/rest/action/document/RestUpdateActionTests.java

```diff
@@ -47,4 +59,44 @@ public void testTypeInPath() {
             .build();
         dispatchRequest(validRequest);
     }
+
+    public void testUpdateDocVersion() {
+        Map<String, String> params = Collections.singletonMap("version", "100");
```

Comment:
Maybe combine the new tests into a single test where the parameters include either version or version_type or both.

Reply:
Yes, I updated the tests.

Reply:
Sorry, I wasn't clear. I meant a single block of test with a randomized parameter. Something like this:

```java
    public void testUpdateDocVersion() {
        Map<String, String> params = new HashMap<>();
        if (randomBoolean()) {
            params.put("version", Long.toString(randomNonNegativeLong()));
            params.put("version_type", randomFrom(VersionType.values()).name());
        } else if (randomBoolean()) {
            params.put("version", Long.toString(randomNonNegativeLong()));
        } else {
            params.put("version_type", randomFrom(VersionType.values()).name());
        }
        String content =
            "{\n" +
                "    \"doc\" : {\n" +
                "        \"name\" : \"new_name\"\n" +
                "    }\n" +
                "}";
        FakeRestRequest updateRequest = new FakeRestRequest.Builder(xContentRegistry())
            .withMethod(RestRequest.Method.POST)
            .withPath("test/_update/1")
            .withParams(params)
            .withContent(new BytesArray(content), XContentType.JSON)
            .build();
        ActionRequestValidationException e = expectThrows(ActionRequestValidationException.class,
            () -> action.prepareRequest(updateRequest, mock(NodeClient.class)));
        assertThat(e.getMessage(), containsString("internal versioning can not be used for optimistic concurrency control. " +
            "Please use `if_seq_no` and `if_primary_term` instead"));
    }
```

Reply:
I see it. Thanks @dnhatn